pkg/routeagent_driver/handlers/calico/ippool_handler.go

```diff
@@ -0,0 +1,256 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package calico
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	calicoapi "github.com/projectcalico/api/pkg/apis/projectcalico/v3"
+	calicocs "github.com/projectcalico/api/pkg/client/clientset_generated/clientset"
+	"github.com/submariner-io/admiral/pkg/log"
+	submV1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/cni"
+	"github.com/submariner-io/submariner/pkg/event"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/rest"
+	k8snet "k8s.io/utils/net"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	submarinerIPPool = "submariner.io/ippool"
+)
+
+type createUpdateFunc func(ipPoolObj *calicoapi.IPPool) error
+
+type calicoIppoolHandler struct {
+	event.HandlerBase
+	restConfig      *rest.Config
+	cs              *calicocs.Clientset
+	remoteEndpoints map[string]*submV1.Endpoint
+	isGateway       bool
+}
+
+var logger = log.Logger{Logger: logf.Log.WithName("CalicoIPPool")}
+
+func NewCalicoIPPoolHandler(restConfig *rest.Config) event.Handler {
+	return &calicoIppoolHandler{
+		restConfig:      restConfig,
+		remoteEndpoints: map[string]*submV1.Endpoint{},
+	}
+}
+
+func (h *calicoIppoolHandler) GetNetworkPlugins() []string {
+	return []string{cni.Calico}
+}
+
+func (h *calicoIppoolHandler) GetName() string {
+	return "Calico_Ippool handler"
+}
+
+func (h *calicoIppoolHandler) Init() error {
+	var err error
+
+	h.cs, err = calicocs.NewForConfig(h.restConfig)
+	if err != nil {
+		return errors.Wrap(err, "error initializing Calico clientset")
+	}
+
+	return nil
+}
+
+func (h *calicoIppoolHandler) RemoteEndpointCreated(endpoint *submV1.Endpoint) error {
+	err := h.handleRemoteEndpointCreateOrUpdate(endpoint, true)
+	if err != nil {
+		return errors.Wrapf(err, "failed to handle RemoteEndpointCreated event")
+	}
+
+	return nil
+}
+
+func (h *calicoIppoolHandler) RemoteEndpointUpdated(endpoint *submV1.Endpoint) error {
+	err := h.handleRemoteEndpointCreateOrUpdate(endpoint, false)
+	if err != nil {
+		return errors.Wrapf(err, "failed to handle RemoteEndpointUpdated event")
+	}
+
+	return nil
+}
+
+func (h *calicoIppoolHandler) RemoteEndpointRemoved(endpoint *submV1.Endpoint) error {
+	delete(h.remoteEndpoints, endpoint.Name)
+
+	if !h.isGateway {
+		logger.Info("Ignore RemoteEndpointRemoved event (node isn't Gateway)")
```

Comment:
I don't think we need this logged by deault.

```suggestion
		logger.V(log.TRACE).Info("Ignore RemoteEndpointRemoved event (node isn't Gateway)")
```